src/legacy/core_plugins/data/public/query/query_bar/components/query_bar.tsx

```diff
@@ -409,6 +329,7 @@ export class QueryBarUI extends Component<Props, State> {
   }
 
   private onLuceneSyntaxWarningOptOut(toast: Toast) {
+    if (!this.props.store) return;
```

Comment:
I don't think that's true, this conditional https://github.com/elastic/kibana/blob/ae5ecfff026c4f0ae7fafba2de62822fab6f2792/src/legacy/core_plugins/data/public/query/query_bar/components/query_bar.tsx#L288 passes if store is undefined or null and then `onLuceneSyntaxWarningOptOut` will be called.

Also, I'm not too familiar with that non-null assertion operation, but it seems to me like it could lead to undetected bugs in the future. Even if that function is not called when store is undefined today, if the code were changed in the future so that it does, typescript would not warn about the issue.

Reply:
ATM `onLuceneSyntaxWarningOptOut` is being called only from within `handleLuceneSyntaxWarning` that already validates `store`.

We just started using it recently. I agree with you, but i usually prefer consistency. But it's a nit. I don't mind.